flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosResourceManagerDriver.java

```diff
@@ -739,172 +591,123 @@ static ResourceID extractResourceID(Protos.TaskID taskId) {
 		}
 	}
 
-	/**
-	 * Creates the Fenzo optimizer (builder).
-	 * The builder is an indirection to facilitate unit testing of the Launch Coordinator.
-	 */
-	private static TaskSchedulerBuilder createOptimizer() {
-		return new TaskSchedulerBuilder() {
-			TaskScheduler.Builder builder = new TaskScheduler.Builder();
-
-			@Override
-			public TaskSchedulerBuilder withLeaseRejectAction(Action1<VirtualMachineLease> action) {
-				builder.withLeaseRejectAction(action);
-				return this;
-			}
-
-			@Override
-			public TaskSchedulerBuilder withRejectAllExpiredOffers() {
-				builder.withRejectAllExpiredOffers();
-				return this;
-			}
+	private static String extractTerminatedDiagnostics(ResourceID id, Protos.TaskStatus status) {
+		return String.format("Worker %s terminated with status: %s, reason: %s, message: %s.",
+				id, status.getState(), status.getReason(), status.getMessage());
+	}
 
-			@Override
-			public TaskSchedulerBuilder withLeaseOfferExpirySecs(long leaseOfferExpirySecs) {
-				builder.withLeaseOfferExpirySecs(leaseOfferExpirySecs);
-				return this;
-			}
+	private static TaskSchedulerBuilder createOptimizer() {
+		return new TaskSchedulerBuilderImpl();
+	}
 
-			@Override
-			public TaskScheduler build() {
-				return builder.build();
-			}
-		};
+	@VisibleForTesting
+	void assertStateCleared() {
+		assert(workersInNew.isEmpty());
+		assert(requestResourceFutures.isEmpty());
```

Comment:
Would it make sense to also check that the actors are cleared (i.e. `MesosResourceManagerDriver::stopSupportingActorsAsync` was performed properly)?

Reply:
This is a production method. It is expected to be called after the actors are created and before any new worker is requested/added.

I guess you mean verifying actors being cleared in `MesosResourceManagerDriverTest::testClearStateAfterRevokeLeadership`? That makes sense to me. We can verify wether `stopActor` are called for all the actors.

Reply:
Exactly. I came across this code segment when going through `MesosResourceManagerDriverTest::testClearStateAfterRevokeLeadership`.